### ISSUE TITLE
ML - Supervised learning

### ISSUE THREAD

**Issue by yoonghan** [yoonghan/yoonghan.github.io#505]
## Type
 
- Regression - split linear charts
- Classification 
  - binary classification = spam/not
  - multiclass classification = dog, cat, giraffe
  - multi label = comedy + action

Training
Training > validation > test

## Examples
- Logistic Regression(Yes/No)
- Linear Regression(House Price)
- Decision Tree.
- Neural Network, CNN, RNN - recognize handwriting

**Comment by yoonghan** [comment]
## Feature engineering
Making data more relevant, like dob change to age

Technique
(Structured data)
- feature extraction - change dob to age
- feature selection - choose important field
- feature transformation - normalise the feature

(Unstructured data)
- text data - convert into numbers, tokens
- image - CNN with edge

**Comment by yoonghan** [comment]
## Root Mean Square Error
Metrics for **regression** by calculating the SQR of averaged square difference between predicted and actual values.

## Mean Absolute Error (MAE)
Measures the average magnitude of errors in a set of predictions without considering the direction? Also for **regression**

## R Square
R-squared is a metric that measures the goodness of fit in regression models. It shows how well the independent variables explain the variance in the dependent variable.

| Metric | Use When | Strength |
| -- | -- | -- |
| RMSE | You want to measure error size in original units | Easy to interpret, sensitive to outliers |
| R² | You want to explain variance explained by the model | Good for comparing models on same dataset |

**Comment by yoonghan** [comment]
## Multi-class classification vs Multi-label classification

Multi-class assigns each instance to one of several possible classes, while multi-label classification assigns each instance to one or more classes

With multi-label classification, you can train models and classify your documents with more than one label. For example, you can use multi-label classification to categorize customer contact transcripts with one or more labels to identify departments within your company like Payments, Renewals, or Tech Support. These labels can then be mapped to relevant content in your support library or directed towards the appropriate contacts within your company. ML models for multiclass classification problems allow you to generate predictions for multiple classes (predict one of more than two outcomes).

Therefore, the key difference is that multi-class classification assigns each instance to one of several possible classes (e.g., an image classified as either a cat, dog, or bird), whereas multi-label classification can assign each instance to multiple classes simultaneously (e.g., a document classified as both "science" and "technology").

**Comment by yoonghan** [comment]
## Discriminative Model Classification vs Generative Model

| Category | Type | Learns to... | Example Models | Learning Type |
| -- | -- | -- | -- | -- |
| Discriminative | Supervised | Model the decision boundary between classes | Logistic Regression, SVM, Random Forest, BERT (fine-tuned), most classifiers | Supervised |
| Generative | Supervised or Unsupervised | Model how the data is generated (i.e., full data distribution) | Naive Bayes, GANs, LDA, Variational Autoencoders | Can be both |

**Comment by yoonghan** [comment]
ImageNet or CIFAR-10 To benchmark images

**Comment by yoonghan** [comment]
## Confusion Matrix
To to evaluate the performance of **classification** models by displaying TRUE positive, TRUE negative, FALSE positive and FALSE negative. Shows how well the model perform
